aws/logs_monitoring/lambda_function.py

```diff
@@ -432,4 +432,36 @@ def parse_service_arn(source, key, bucket):
         #For S3 access logs we use the bucket name to rebuild the arn
         if bucket:
             return "arn:aws:s3:::" + bucket
+    if source == "cloudfront":
+        #For Cloudfront logs we need to get the account and distribution id from the lambda arn and the filename
+        #1. We extract the cloudfront id  from the filename
+        #2. We extract the AWS account id from the lambda arn
+        #3. We build the arn
+        namesplit = key.split("/")
+        if len(namesplit) > 0:
+            filename = namesplit[len(namesplit)-1] 
+            #(distribution-ID.YYYY-MM-DD-HH.unique-ID.gz)
+            filenamesplit = filename.split(".")
+            if len(filenamesplit) > 3:
+                distributionID = filenamesplit[len(filenamesplit)-4].lower()
```

Comment:
filenamesplit[0] ?

Reply:
Yes.
The file path is `bucket-name.s3.amazonaws.com/optional-prefix/distribution-ID.YYYY-MM-DD-HH.unique-ID.gz`

So the first thing we do is to extract `distribution-ID.YYYY-MM-DD-HH.unique-ID.gz` and then we get the `distribution-ID` in the first element.